### PR TITLE
Re/feat/components connect db

### PR DESCRIPTION
**Pull request by r3ym3l**
## What does this PR do?
- Connected the text box components to the database. The displayed data can also be deleted now using the text box delete button.

### Screenshots (optional)
image

### Steps to test/review (optional)
- Run back-end: `npm start`
  - Make sure you have the necessary back-end .env file.
- Build the extension: `npm run build`
  - Go to chrome and navigate to _Manage extensions_
  - Make sure _Developer mode_ is on
  - Load build file
- Run front-end: `npm start`
- Open MongoDB Atlas to look at current data and compare it to what is being shown on web app.
- Add a text using chrome extension (highlight text and right-click mouse to add to workspace)
- Refresh both MongoDB and the web app to see if the new data was added and displayed.
- Click on the source button to see whether it will take you back to the source of the text that was just added.
- Click on the delete button and then check MongoDB to see if the text is gone.

## Checklist before merging
- [x] Applied options on the right:
      - Reviewers
      - Assignee
      - Labels
      - Project
      - Milestone
      - Linked Issues
- [x] All linter rules passed
- [x] Auto-build success (TBA)

### CODE REVIEW

**Review by mariellemanlulu — COMMENTED** [review]
Fetching data works!
---
1. I'm getting an error upon loading front-end. its because we had "bad data" where creation date is still s tring. I think it will fix it if we delete that data.
Screenshot 2022-03-04 123223

2. Delete doesn't work for me. Here is the error message. Am I missing something?
Screenshot 2022-03-04 122945

**Comment by mariellemanlulu** [comment]
Tried again, the issue was I was using the old server code while adding text, so it was saving the old text object properties. 

Works fine now! -- approving :)

**Review by mariellemanlulu — APPROVED** [review]

File: front-end/src/atoms/TextBox.js [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
+
 
 const MAX_CHARACTERS = 142;
 
 
-function TextBox({text, source}) {
+function TextBox({textItem}) {
+  const [ texts, setTexts ] = useState([]);
+
+  const handleDelete = (id) => {
```
Comment (deleonkf) [inline front-end/src/atoms/TextBox.js:16]: Might be able to push this handle function (plus the add function in TextBoxAdd.js) up to the parent (TextList.js). That way, any add and delete will automatically re-render the page

File: front-end/src/App.js [hunk 2]
```diff
@@ -36,7 +39,27 @@ const mockData = [
...
+  const [ dataIsLoaded, setDataIsLoaded ] = useState(false);
   const { isAuthenticated } = useAuth0();
+  const [ textItems, setTextItems ] = useState(null);
+
+  /**
+   * Grabs data from the db.
+   * 
+   * @returns 				returns data from the db. 
+   */
+  async function getTexts() {
```
Comment (deleonkf) [inline front-end/src/App.js:51]: Can move this function down to TextList.js

**Review by deleonkf — APPROVED** [review]

**Pull request merged**